model/task_lifecycle.go

```diff
@@ -578,6 +577,10 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 		}
 	}
 
+	if t.ResetWhenFinished && !t.IsPartOfDisplay() && !t.IsPartOfSingleHostTaskGroup() {
```

Comment:
This adds the ability for tasks that aren't display tasks or single-host task groups to have a delayed restart (needed for delayed restarting of aborted tasks)